include/core/CStateDecompressor.h

```diff
@@ -94,20 +94,16 @@ class CORE_EXPORT CStateDecompressor : public CDataSearcher {
         struct SBaseBoostJsonHandler {
 
             //! The maximum number of elements allowed in an array
-            static constexpr std::size_t max_array_size =
-                std::numeric_limits<std::size_t>::max();
+            static constexpr std::size_t max_array_size = 1000000; // 1 milion elements
```

Comment:
Are these numbers always the same as for `CJsonStateRestoreTraverser`? If so, should they be moved to a shared location to avoid code duplication? (`constants.h` or so?)